docs/content/docs/reference/react/hooks.mdx

```diff
@@ -0,0 +1,393 @@
+- `isSuccess: boolean` returns `true` if the magic link email user has successfully send.
+- `isError: boolean` returns `true` if an error occurred.
+- `error?: {status: number, error: string, message: string}` provides details about the error.
+- `PasswordlessOptions`:
+  - `locale?: string` locale of the user, in two digits, for instance `en`.
+  - `allowedRoles?: string[]`: allowed roles of the user. Must be a subset of the default allowed roles defined in Hasura Auth.
+  - `defaultRole?: string`: default role of the user. Must be part of the default allowed roles defined in Hasura Auth.
+  - `displayName?: string`
+  - `metadata?: Record<string, unknown>`: Custom additional user information stored in the `metadata` column.
+  - `redirectTo?: string`: redirection path in the client application that will be used in the link in the verification email. For instance, if you want to redirect to `https://myapp.com/success`, the `redirectTo` value is `'/success'`.
+
+#### Usage
+
+```jsx
+import { useState } from 'react'
+import { useEmailPasswordlessSignIn } from '@nhost/react'
+
+const Component = () => {
+  const [email, setEmail] = useState('')
+  const { signIn, isLoading, isSuccess, isError, error } = useEmailPasswordlessSignIn(email)
+  return (
+    <div>
+      <input value={email} onChange={(event) => setEmail(event.target.value)} placeholder="Email" />
+      <button onClick={signUp}>Register</button>
+      {isSuccess && (
+        <div>
+          An email has been sent to {email}. Please check your mailbox and click on the
+          authentication link.
+        </div>
+      )}
+    </div>
+  )
+}
+```
+
+### Sign out
+
+The `useSignOut` hook accepts an `all` argument that will be used when the `signOut` method will be called. This value can be overriden in calling `signOut(allValue)`.
+
+```js
+const { signOut, isSuccess } = useSignOut(all?: boolean)
+```
+
+- `signOut: (all?: boolean) => void` will log the current user out. If `all` is set to true, it will deactivate the active session from all other devices. The `all` argument will take precedence over the the possible state value used when creating the hook.
+- `isSuccess: boolean` returns `true` when the user has successfully signed out.
+
+#### Usage
+
+```jsx
+import { useState } from 'react'
+import { useSignOut, useAuthenticated } from '@nhost/react'
+
+const Component = () => {
+  const { signOut, isSuccess } = useSignOut()
+  const authenticated = useAuthenticated()
+  if (authenticated)
+    return (
+      <div>
+        <button onClick={signUp}>Sign out</button>
+        {isSuccess && <div>You have successfully signed out!</div>}
+      </div>
+    )
+  else return <div>Not authenticated</div>
+}
+```
+
+## Authentication status
+
+### `useLoading`
+
+<!-- TODO better documentation -->
+
+During startup, the client may need some initial steps to determine the authentication status, like fetching a new JWT from an existing refresh token.
```

Comment:
"The client may need some initial steps to determine the authentication status during startup, like fetching a new JWT from an existing refresh token."